docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,294 @@
+            localVariable += Library_ExternalFunction( pArgument );
+
+            LogDebug( "Current value is %d.", ( int ) localVariable );
+        }
+
+        if( localVariable < 0 )
+        {
+            LogWarn( "Failed to calculate positive value." );
+        }
+
+        LogInfo( "Calculation done." );
+    }
+
+    return status;
+}
+
+
+/* A separator is placed between all function implementations. */
+/*-----------------------------------------------------------*/
+
+/* Implementations of application-facing functions are at the bottom of the file. */
+
+bool Library_ApplicationFunction( void ) /* Functions with no arguments have void in their argument list. */
+{
+    LIBRARY_FUNCTION_MACRO( globalArray );
+
+    return true;
+}
+
+/* Separator and newline at end of file */
+/*-----------------------------------------------------------*/
+
+@endcode
+
+@section guide_developer_styleguide_naming Naming
+@brief Naming convention used in this SDK.
+
+The naming convention aims to differentiate this SDK's files, variables, and functions to avoid name collisions. In general:
+- The first characters of all <i>publicly visible</i> names should identify the name as part of this SDK. <br>
+  <b>Example:</b> For general-purpose libraries (such as MQTT), names start with <library-name>_ (i.e. MQTT_ for the MQTT API).
+- Words in names should be ordered with the most general word first and the most specific word last. <br>
+  <b>Example:</b> `core_mqtt_serializer.c` identifies a file as part of the general MQTT library. `MQTT_Connect` identifies a public-facing function of the general MQTT library.
+- Names should avoid using abbreviations.
+
+@subsection guide_developer_styleguide_naming_definedconstantsandenumvalues Defined constants and enum values
+@brief Naming convention for constants set using preprocessor @c #`define` and enum values.
+
+@formattable{defined constants and enum values}
+@formattableentry{<b>LibraryDescription</b>,Defined constants and enum values in application-facing library header files,`MQTTSuccess` (core_mqtt_lightweight.h)}
+@formattableentry{<b>LIBRARY_DESCRIPTION</b>,Defined constants and enum values in demos and tests,`MQTT_EXAMPLE_TOPIC`}
+@formattableentry{<b>DESCRIPTION</b>,Internal constants and enum values<br>No `AWS_` prefix for internal names in AWS-specific libraries,`ROOT_CA_CERT_PATH`}
+
+@subsection guide_developer_styleguide_naming_files Files
+@brief Naming convention for files.
+
+@formattable{files}
+@formattableentry{<b>library</b>`_`<b>description</b>`.extension`,General library file,`core_mqtt_serializer.c`}
+@formattableentry{<b>library</b>`_internal.h`,Internal library header,`core_mqtt_internal.h`}
+@formattableentry{<b>library</b>`_`<b>demo</b>`_description.c`,Library demo source,`mqtt_demo_plaintext.c`}
+@formattableentry{<b>library</b>`_description_`<b>utest</b>`.c`<br><b>library</b>`_description_`<b>system_test</b>`.c`,Library test source,`core_mqtt_utest.c`<br>`mqtt_system_test.c`}
```

Comment:
The description said some the optional parts of this text were *italic*? If it's not, then we can remove that part.